tests/testthat/test-convert_rle_to_df.R

```diff
@@ -0,0 +1,63 @@
+prepared_unreliable_regions <- function() {
+  return(
+    combined_unreliable_regions %>%
+      dplyr::filter(`1_4_0` == TRUE) %>%
+      dplyr::select(Start, End, `Virus segment`, Categories)
+  )
+}
+
+rles <- function() {
+  return(
+    rles <- convert_bam_to_rle(here::here("tests", "files", "reads_1.sorted.bam"))
+  )
+}
+
+test_that("run_with_defaults", {
+  expect_snapshot(
+    convert_rle_to_df(rles())
+  )
+})
+
+test_that("run_without_unreliable_regions", {
+  expect_snapshot(
+    convert_rle_to_df(
+      rles(), unreliable_regions = prepared_unreliable_regions()
+    )
+  )
+})
+
+test_that("unreliable_regions_has_effect", {
+  expect_failure(
+    expect_equal(
+      convert_rle_to_df(
+        rles()
+      ),
+      convert_rle_to_df(
+        rles(),
+        unreliable_regions = prepared_unreliable_regions(),
+      )
+    )
+  )
```

Comment:
Should check to make sure the unreliable regions are also being used correctly. As is only checks if any change is happening